File: nx-gradle/spring-boot/spring-boot-app/src/main/java/com/example/springboot/springbootapp/HelloController.java
package com.example.springboot.springbootapp;

import com.example.springboot.springbootlib.HelloService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HelloController {

  @Autowired
  private HelloService helloService;

  @GetMapping("/")
  public String greeting() {
    return helloService.message();
  }
}

File: nx-gradle/spring-boot/spring-boot-kt-app/src/main/kotlin/com/example/springboot/springbootktapp/HelloController.kt
package com.example.springboot.springbootktapp

import com.example.springboot.springbootktlib.HelloService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HelloController(
  @Autowired val helloService: HelloService,
) {

    @GetMapping("/")
    fun greeting():String = helloService.message()

}
